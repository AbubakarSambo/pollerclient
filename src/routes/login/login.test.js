import LoginForm from './Forms/Login.forms';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import React from 'react';

test('Login form renders correctly', () => {
    const rendered = renderer.create(
        <LoginForm />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});

test('Login form calls submit with username and password values when clicked', () => {
    const signIn = jest.fn();
    const wrapper = mount(
        <LoginForm signIn={signIn} />
    );
    wrapper.find('#username').simulate('change', { target: { name: 'username', value: 'fsadmin' } });
    wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'password' } });

    const signInButton = wrapper.find('button');
    signInButton.simulate('click');
    expect(signIn).toBeCalledWith({ username: 'fsadmin', password: 'password' });
});