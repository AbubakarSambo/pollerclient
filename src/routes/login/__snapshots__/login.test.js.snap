// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form renders correctly 1`] = `
<div>
  <div
    className="sc-kpOJdX hqChUe"
    label="Username"
    required={true}
    type="text"
  >
    <span
      className="sc-jzJRlG kQupf"
    >
      Username
    </span>
    <input
      id="username"
      onChange={[Function]}
      placeholder="Username"
      value=""
    />
  </div>
  <div
    className="sc-kpOJdX ddVwRn"
    label="Password"
    required={true}
    type="password"
  >
    <span
      className="sc-jzJRlG kQupf"
    >
      Password
    </span>
    <input
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <div
    className="sc-gzVnrw cIYwRj"
    default="100%"
  >
    <button
      className="sc-bdVaJa eenbRK"
      id="signInButton"
      onClick={[Function]}
    >
      Sign in
    </button>
  </div>
</div>
`;
